core/server/services/auth/api-key/admin.js

```diff
@@ -145,13 +145,15 @@ const authenticateWithToken = (req, res, next, {token, JWT_OPTIONS}) => {
 
         if (apiKey.get('user_id')) {
             // fetch the user and store it on the request for later checks and logging
-            models.User.findOne({id: apiKey.get('user_id')}).then((user) => {
-                req.user = user;
+            return models.User.findOne({id: apiKey.get('user_id')}).then((user) => {
+                // Personal Tokens shouldn't work for users that aren't active
+                if (user.isActive()) {
```

Comment:
Passing `id: blah, status: 'active'` to findOne would also work here - whatever you think is best though!

Reply:
Oh and I just realised, `findOne` will resolve with `null` if the user cannot be found, so this will error in that case.

You have a few options, you can pass `require: true` in the options (https://bookshelfjs.org/api.html#Model-instance-fetch) which'll cause the promise to reject - or you can check for the user before using the object :relaxed:

Reply:
I'm going to keep this implementation because it covers all active states that are listed there: https://github.com/TryGhost/Ghost/blob/62f6d35afc233b9d285a3e5ad6ae5821ba88f9ed/core/server/models/user.js#L14

Unless the `warn` states are deprecated @allouis ?

Reply:
Nope not deprecated, but `status: active` also takes into account those states, just FYI! https://github.com/TryGhost/Ghost/blob/master/core/server/models/user.js#L416-L417

Totally fine to leave as is!

Reply:
Good to know it's taken into account :exploding_head: :tada: 
I'll go the `require: true` way then!